components/Notifications/Notifications.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+
+import data from "../../data/notifications.json";
+
+const Banner = ({
+  type,
+  description,
+  date,
+  isOpen,
+  isMultiple,
+  handleDismissAll,
+  allDismissed,
+}: any) => {
+  const [isDismissed, setIsDismissed] = React.useState(true);
+
+  function handleDismiss() {
+    setIsDismissed(true);
+
+    // saves notification to local storage
+    let datesArray = JSON.parse(localStorage.getItem("notifications")) || [];
+    datesArray.push(date);
+    localStorage.setItem("notifications", JSON.stringify(datesArray));
+  }
+
+  React.useEffect(() => {
+    const datesArray = JSON.parse(localStorage.getItem("notifications")) || [];
+    if (allDismissed || datesArray.includes(date)) setIsDismissed(true);
+    else setIsDismissed(false);
+  }, [allDismissed, date]);
+
+  return (
+    <div
+      className={`${
+        (isDismissed || isOpen) && "translate-y-full"
+      } ease fixed inset-x-0 bottom-0 z-20 transform transition duration-300 sm:flex sm:justify-center sm:pb-4`}
+    >
+      <div className="flex items-center justify-between gap-x-6 bg-cesium px-6 py-2.5 sm:rounded-2xl sm:py-3 sm:pl-4 sm:pr-3.5 sm:shadow-md">
+        <div className="select-none text-sm leading-6 text-white">
+          <a>
+            <strong className="font-semibold">
+              <i className="bi bi-info-circle-fill"></i> {type}
+            </strong>
+            <svg
+              viewBox="0 0 2 2"
+              className="mx-2 inline h-0.5 w-0.5 fill-current"
+              aria-hidden="true"
+            >
+              <circle cx={1} cy={1} r={1} />
+            </svg>
+            {description}
+          </a>
+        </div>
+        <div className="flex justify-between gap-x-5 sm:gap-x-3">
+          {isMultiple && (
+            <>
+              <button
+                title="Dismiss All"
+                type="button"
+                className="-m-1.5 mr-0.5 flex-none"
+                onClick={() => handleDismissAll()}
+              >
+                <i className="bi bi-bell-slash text-white"></i>
+              </button>{" "}
+            </>
+          )}
+          <button
+            title="Dismiss"
+            type="button"
+            className="-m-1.5 flex-none p-1.5"
+            onClick={() => handleDismiss()}
+          >
+            <i className="bi bi-x-circle-fill text-white" />
+          </button>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const Notifications = ({ isOpen }) => {
+  const [allDismissed, setAllDismissed] = React.useState(false);
+
+  const notifications = data;
```

Comment:
You need to intersect this with the already saved notifications (those in local storage). Otherwise, it will always show the "Dismiss all" button whether there's only one notification to see or not.

Reply:
I'm not sure I understood what you're saying... The notifications are filtered before being mapped to the banner elements and the notifications that are already in local storage are filtered OUT of the mapping. Furthermore it makes sure that the last element does not include a "Dismiss All" button with the condition `if (index === 0) isMultiple = false;` inside the map. If there's only one notification, the element will naturally have index of 0 and not display the "Dismiss All" icon.

Reply:
@RuiL1904 in the meanwhile, improved the way the code works and also made only the "top" notification show, when the top notification is dismissed the one under that will appear and so on. i don't know why i didn't think of this but before if we had like two notifications of different sizes we could see the bottom notifications behind the top one and it looked super bad.

Reply:
I was, in fact, not well-testing it. But now I think I've encountered a new problem. Steps to reproduce:

1. Create two notifications that will be displayed (within the past 7 days)
2. Open Calendarium
3. Dismiss the first notification, but not the second one
4. Close Calendarium and open it on a new tab
5. Notice the second notification does not appear, despite its date not being on local storage

Can you reproduce? 🤔

Reply:
Yes, I was able to recreate. Looking into it now ;)

Reply:
@RuiL1904 Should be fixed! :pray: